### ISSUE TITLE
Bug: [no-unused-vars] "vars": "local" not working

### ISSUE THREAD

**Issue by zacaj** [eslint/eslint#20376]
### Environment

https://eslint.org/play with default settings

### What parser are you using?

Default (Espree)

### What did you do?

```js
/*eslint no-unused-vars: ["error", { "vars": "local" }]*/

let globalVar = "b";

function globalFunc() {
  const localVar= 3;
}
```

<img width="570" height="300" alt="Image" src="https://github.com/user-attachments/assets/cfa193d7-dbc1-4f95-ba19-4648b585edbf" />

### What did you expect to happen?

 globalVar and globalFunc shouldn't trigger errors

### What actually happened?

 globalVar and globalFunc trigger errors regardless of `vars` setting

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tdW51c2VkLXZhcnM6IFtcImVycm9yXCIsIHsgXCJ2YXJzXCI6IFwibG9jYWxcIiB9XSovXG5cbmxldCBnbG9iYWxWYXIgPSBcImJcIjtcblxuZnVuY3Rpb24gZ2xvYmFsRnVuYygpIHtcbiAgY29uc3QgbG9jYWxWYXI9IDM7XG59Iiwib3B0aW9ucyI6eyJydWxlcyI6e30sImxhbmd1YWdlT3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

off topic but thank you for having a "I am willing to submit a pull request for this issue." checkbox!  wish every project had that

**Comment by mdjermanovic** [comment]
Hi @zacaj, thanks for the issue! This work as intended because with the default `sourceType: "module"` variables `globalVar` and `globalFunc` are not global but scoped to the module. Here's the same example with `sourceType: "script"` set in configuration ("Versioning and Config" in the Playground) where `globalVar` and `globalFunc` are indeed global and thus not reported by this rule:

https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tdW51c2VkLXZhcnM6IFtcImVycm9yXCIsIHsgXCJ2YXJzXCI6IFwibG9jYWxcIiB9XSovXG5cbmxldCBnbG9iYWxWYXIgPSBcImJcIjtcblxuZnVuY3Rpb24gZ2xvYmFsRnVuYygpIHtcbiAgY29uc3QgbG9jYWxWYXI9IDM7XG59Iiwib3B0aW9ucyI6eyJydWxlcyI6e30sImxhbmd1YWdlT3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319LCJzb3VyY2VUeXBlIjoic2NyaXB0In19fQ==

**Comment by zacaj** [comment]
the setting is 'all' or 'local', not 'all' or 'non-global'.   'module' level declarations are not really local per MDN.

"When you declare a variable *outside of any function, it is called a global variable*, because it is available to any other code *in the current document*. When you declare *a variable within a function, it is called a local variable*, because it is available only within that function."

'local' should only apply to variables within functions.    If the `vars` setting had a third option for module level vs global this would make more sense; as is this setting is pretty much useless in modern module based code since true 'global' variables are discouraged to begin with.

**Comment by mdjermanovic** [comment]
This option was added to allow variables in the global scope because they can be used in other scripts (https://github.com/eslint/eslint/issues/367). Variables defined in the module scope can only be used in that module so they have no purpose otherwise. 

I think the [docs](https://eslint.org/docs/latest/rules/no-unused-vars#vars) are already clear that `"all"`/`"local"` relates to variables in the global scope:

> The `vars` option has two settings:
> 
> * `all` checks all variables for usage, including those in the global scope. However, it excludes variables targeted by other options like `args` and `caughtErrors`. This is the default setting.
> * `local` checks only that locally-declared variables are used but will allow global variables to be unused.

but I'll submit a PR to reword this to avoid confusion.

Either way, the `"local"` option is definitely not intended to allow variables in the module scope.
